.github/workflows/test.yml

```diff
@@ -22,19 +22,31 @@ jobs:
   system-test:
     runs-on: ubuntu-latest
     steps:
+      - uses: actions/checkout@v2
+      
       - name: Extract branch name
+        id: extract_branch
         shell: bash
         run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
-        id: extract_branch
 
-      - uses: actions/checkout@v2
+      - name: Determine the default stage for this run
```

Comment:
this determines the right stage to run against depending on this was run from a workflow dispatch event or as a branch event